core/src/main/java/org/apache/calcite/rel/metadata/JaninoRelMetadataProvider.java

```diff
@@ -475,33 +356,15 @@ private static StringBuilder paramList(StringBuilder buff, Method method) {
 
   synchronized <M extends Metadata, H extends MetadataHandler<M>> H revise(
       Class<? extends RelNode> rClass, MetadataDef<M> def) {
-    if (ALL_RELS.add(rClass)) {
-      HANDLERS.invalidateAll();
-    }
     //noinspection unchecked
     return (H) create(def);
   }
 
   /** Registers some classes. Does not flush the providers, but next time we
    * need to generate a provider, it will handle all of these classes. So,
    * calling this method reduces the number of times we need to re-generate. */
+  @Deprecated
   public void register(Iterable<Class<? extends RelNode>> classes) {
-    // Register the classes and their base classes up to RelNode. Don't bother
```

Comment:
Is there any loss of functionality difference with this patch (difficult to infer from the dynamically generated code)? Are RelNode classes that are defined by engines relying on Calcite, e.g., Hive, handled as before? Will this require code changes in those systems?

Reply:
There is no loss in functionality.